### ISSUE TITLE
readonly modifier is ignored in unions

### ISSUE THREAD

**Issue by aleksey-bykov** [Microsoft/TypeScript#9108]
```typescript 
interface X {
    readonly value: 'x';
}
interface Y {
    value: 'y';
}
type Z = X | Y;
var z: Z = <Z>{ value: 'y' };
z.value = 'x'; // expected an error saying that value: 'x' is readonly and cannot be set, actual OK
```

UPDATE:

even this example doesn't work (with both possible options being `readonly`)

```typescript 
interface X {
    readonly value: 'x';
}
interface Y {
    readonly value: 'y';
}
type Z = X | Y;
var z: Z = <Z>{ value: 'y' };
z.value = 'x'; // expected an error saying that value: 'x' is readonly and cannot be set, actual OK
```

**Comment by sandersn** [comment]
#9167 fixes this for unions and intersections.
